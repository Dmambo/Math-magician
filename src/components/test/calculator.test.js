// import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Calculator from '../calculator';

// test calculator with snapshot

test('Calculator renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Calculator />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
