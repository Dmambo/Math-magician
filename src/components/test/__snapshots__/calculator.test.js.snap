// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div
  className="calculate"
>
  <div
    className="math"
  >
    <h1>
      Let's do some math!
    </h1>
  </div>
  <div
    className="calculator"
  >
    <div
      className="calculator__display"
    >
      <div
        className="calculator__display__curr"
      >
        0
      </div>
    </div>
    <div
      className="calculator__buttons"
    >
      <div
        className="calculator__buttons__row"
      >
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="calculator__buttons__row__button orange"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="calculator__buttons__row"
      >
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="calculator__buttons__row__button orange"
          onClick={[Function]}
          type="button"
        >
          *
        </button>
      </div>
      <div
        className="calculator__buttons__row"
      >
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="calculator__buttons__row__button orange"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="calculator__buttons__row_"
      >
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="calculator__buttons__row__button orange"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="calculator__buttons__row"
      >
        <button
          className="calculator__buttons__row__button zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="calculator__buttons__row__button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="calculator__buttons__row__button orange"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
